For select fields, set model value on initial render

goldenapples opened the pull request:
Select fields are a bit of a special case as far as attribute fields go, in that the select element's value in the UI can appear to be set without the model value itself being defined. This change handles cases where there's no empty value in the select field, so the first option should function as a default value.

When a select field is rendered, if the model's value is empty and the first element isn't an empty option, this will set the model's value to the value of the option which is visibly selected, to avoid confusion.

Fixes #668

goldenapples committed:
For select fields, set model value on initial render

Select fields are a bit of a special case as far as attribute fields go,
in that the select element's value in the UI can appear to be set
without the model value itself being defined. This change handles cases
where there's no empty value in the select field, so the first option
should function as a default value.

When a select field is rendered, if the model's value is empty and the
first element isn't an empty option, this will trigger an initial
"change" event to set the model's value to the value of the option which
is visibly selected, to avoid confusion.

goldenapples committed:
Add test specs for select field default functionality

mattheu committed:
Add comment

goldenapples commented on js/build/shortcode-ui.js:
```diff
@@ -1144,6 +1144,17 @@ var editAttributeField = Backbone.View.extend( {
+			}
+		}
+
+		this.$el.html( this.template( data ) );
```
Where's this line coming from? It doesn't seem to be in the `src` file and it duplicates L1146 above...

mattheu replied:
recompiled to remove

mattheu committed:
Recompile

goldenapples committed:
Handle options with empty values in select fields (#675)

* Handle options with empty values in select fields

If an option with a null value is present, then let the field value
default to empty, rather than to the first visible option value. Also
sets that value on the data object so that it shows as visibly selected.

* Remove commented-out code from specs
